File: examples/inline/typescript/integrations/windsor-ai/002-use-with-agent.ts
import { LlmAgent, MCPToolset } from "@google/adk";

const WINDSOR_API_KEY = "YOUR_WINDSOR_API_KEY";

const rootAgent = new LlmAgent({
    model: "gemini-flash-latest",
    name: "windsor_agent",
    instruction: "Help users analyze their marketing and business data.",
    tools: [
        new MCPToolset({
            type: "StreamableHTTPConnectionParams",
            url: "https://mcp.windsor.ai",
            transportOptions: {
                requestInit: {
                    headers: {
                        Authorization: `Bearer ${WINDSOR_API_KEY}`,
                    },
                },
            },
        }),
    ],
});

export { rootAgent };

File: examples/inline/python/integrations/postman/002-use-with-agent.py
from google.adk.agents import Agent
from google.adk.tools.mcp_tool import McpToolset
from google.adk.tools.mcp_tool.mcp_session_manager import StreamableHTTPConnectionParams

POSTMAN_API_KEY = "YOUR_POSTMAN_API_KEY"

root_agent = Agent(
    model="gemini-flash-latest",
    name="postman_agent",
    instruction="Help users manage their Postman workspaces and collections",
    tools=[
        McpToolset(
            connection_params=StreamableHTTPConnectionParams(
                url="https://mcp.postman.com/mcp",
                # (Optional) Use "/minimal" for essential tools only
                # (Optional) Use "/code" for code generation tools
                # (Optional) Use "https://mcp.eu.postman.com" for EU region
                headers={
                    "Authorization": f"Bearer {POSTMAN_API_KEY}",
                },
            ),
        )
    ],
)